darts/models/kalman.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+import pandas as pd
+
+from abc import ABC
+
+from typing import Optional
+from filterpy.kalman import KalmanFilter
+
+from .filtering_model import FilteringModel
+from ..timeseries import TimeSeries
+
+
+class Kalman(FilteringModel, ABC):
+
+    def __init__(
+            self, 
+            dim_x: int = 1, 
+            dim_z: int = 1, 
+            x = np.array([[1.]]),
+            F = np.array([[1]]),
+            H = np.array([[1]]),
+            P = np.array([[1000.,    0.], [   0., 1000.] ]),
+            Q = 0.1,
+            kf: Optional[KalmanFilter] = None
+        ):
+        """ Kalman filter model
+        This model doesn't predict the future values, instead it predicts 
+        more precise values compared to noise interfered data
+        
+        Parameters
+        ----------
+        dim_x : int
+            Size of the Kalman state vector
+        dim_y : int
+            Size of the measurement vector
+        x : numpy array
+            filter state estimate
+        F : numpy array
+            state transistion matrix
+        H : numpy array
+            measurement function
+        P : numpy array
+            covariance matrix
+        Q : float
+            Process uncertainty/noise multiplicator
+        kf : KalmanFilter
+            Instance of Kalman Filter (for advanced configuration)
+        """
+        super().__init__()
+        if kf is None:
+            self.dim_x = dim_x
+            self.dim_z = dim_z
+            self.kf = KalmanFilter(dim_x=dim_x, dim_z=dim_z)
```

Comment:
Shouldn't all of this be done on `predict()` ?

Reply:
This are all initial parameters.  Or am I not getting something?

Reply:
What I meant is that `self.kf` will contain some state, and this state should be reset every time we want to re-fit the model on a new series (the hyper parameters can be kept, but the model should be re-init). Similar e.g. to what we are doing with ARIMA, ExponentialSmoothing etc.

I'm saying `predict()` (and not `fit`) because I don't think we need a `fit` function for `SmoothingModel`s. (Also I'm having second thoughts I think `FilteringModel` with a `filter()` function [and not a `predict()` one] might have been better in fact 😅 )

Reply:
ok, understood.